src/components/KTable/ColumnVisibilityMenu.vue

```diff
@@ -0,0 +1,91 @@
+<template>
+  <div class="k-table-column-visibility-menu">
+    <KDropdown>
+      <KTooltip text="Show/Hide Columns">
+        <KButton
+          appearance="secondary"
+          class="menu-button"
+        >
+          <template #icon>
+            <CogIcon />
+          </template>
+        </KButton>
+      </KTooltip>
+
+      <template #items>
+        <KDropdownItem
+          v-for="col in columns"
+          :key="col.key"
+          class="column-visibility-menu-item"
+          @click.stop="visibilityMap[col.label] = !visibilityMap[col.label]"
+        >
+          <KCheckbox
+            v-model="visibilityMap[col.label]"
+            :label="col.label"
+          />
+        </KDropdownItem>
+        <KDropdownItem>
+          <KButton
+            appearance="tertiary"
+            class="apply-button"
+            @click="handleApply"
+          >
+            Apply
+          </KButton>
+        </KDropdownItem>
```

Comment:
Apply button doesn't need to be wrapped by KDropdownItem (makes it look weird)

![Screenshot 2024-04-05 at 10 21 29 AM](https://github.com/Kong/kongponents/assets/36751160/554dd65c-c83f-464e-bb76-dd552c336c98)